mm2src/coins/lp_coins.rs

```diff
@@ -3561,6 +3572,20 @@ impl MmCoinEnum {
     pub fn is_eth(&self) -> bool { matches!(self, MmCoinEnum::EthCoin(_)) }
 
     fn is_platform_coin(&self) -> bool { self.ticker() == self.platform_ticker() }
+
+    pub fn maker_secret_hash_algo_v2(&self) -> SecretHashAlgo {
+        match self {
+            MmCoinEnum::EthCoin(_) => SecretHashAlgo::SHA256,
+            _ => SecretHashAlgo::DHASH160,
+        }
+    }
+
+    pub fn taker_secret_hash_algo_v2(&self) -> SecretHashAlgo {
+        match self {
+            MmCoinEnum::EthCoin(_) => SecretHashAlgo::SHA256,
+            _ => SecretHashAlgo::DHASH160,
+        }
+    }
```

Comment:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/d3e6ae140c6866ea7bac661806e8cf3b2df28faa/mm2src/mm2_main/src/lp_swap.rs#L1630-L1639

I was guided by what we have in `detect_secret_hash_algo`, as I see, it can be important is it maker or taker. Or did I misunderstand?

Reply:
Well, this was overly optimized by me I guess. For lightning taker we can use `SHA256` with no problems. Let's make it not depend on taker or maker for v2 :)

Reply:
> Well, this was overly optimized by me. For lightning taker we can use `SHA256` with no problems. Let's make it not depend on taker or maker for v2 :)

Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/673cd11062c5b48d5822bcc505b0949860862cd1